### ISSUE TITLE
Shelving units deleting items

### ISSUE THREAD

**Issue by kaosjr** [cout970/Magneticraft-API-and-Issues#84]
It seems that the shelving items are randomly deleting items. I filled up the entire bottom of one of my shelving units and when I checked on it later, the bottom 9 rows were empty.
I am playing on Material Energy^Natural Capital version 0.7.2, the modlist is on CurseForge.

**Comment by Alexismachina** [comment]
Are you piping stuff in the shelving unit?

**Comment by kaosjr** [comment]
No, I am not. I filled the bottom shelf completely and when O checked it later the bottom 9 rows were empty.

**Comment by phrostt** [comment]
Can confirm have had shelving units eat countless items before I caught on.
One thing that I noticed, at times deleting the shelving unit drops all of the "missing" items on the ground, but as a whole they are very wonky.

**Comment by Alexismachina** [comment]
Yes it was confirmed by Wyld , Bacon_Donnut , Bevo , Everywone who play Bevo Tech Pack and SkyFactory 2.5 ...

**Comment by otakucode** [comment]
I just got bit by this too in Bevo's Tech Pack Reborn, Beta N.  Thought I was just going mad at first, but just had an item I knew was in there disappear so I came looking for a way to report the issue.  I will also mention that I had the same situation as kaosjr, in that I had filled the bottom shelf specifically, and it was from the bottom shelf that the item disappeared.  I had added items after the one I wanted to retrieve, and those items were still there, so it doesn't seem to be just deleting the latest items.

**Comment by Funwayguy** [comment]
I was helping Darkosto and Wyld work on SkyFactory 2.5 and we've identified the cause of the issue. It appears you're using Bytes to store the slot indexes of the shelves in NBT which maxes out at 128. Any item stored after that gets wiped because of the resulting overflow. Please rewrite the tile entity's save handler to use Integers instead of Bytes to store item slot indexes.

**Comment by MechWarrior99** [comment]
@Alexismachina Can you come on IRC and be on 1.7.10 bug fixing duty? I really hate people not being able to use MgC because of bugs. Please?

**Comment by braddr** [comment]
Has anything come from this?  It seems like a shame that a bug, with at least the cause of the problem understood, hasn't resulted in forward progress towards a fixed release.
